### DISCUSSION TITLE
[QUESTION] - Is there feature parity with the Rosetta extension?

### DISCUSSION THREAD

**Discussion by mprather** [opencaesar/platform#15] [Q&A]
## Description
How does the VSCode extension differ from the Eclipse extension?   Or is there feature parity?

**Comment by melaasar** [comment]
The VS Code extension allows one to edit the *.oml files with feature fledged content-assist. One can also use VS Code Gradle plugin to invoke the Gradle tasks of the project to build, query, the ontologies.

The Eclpse extension has File -> New -> Project/Ontology wizards that VSCode one does not currently have.

Also, the VS Code extension allows visualizing OML models in diagram automatically (open an OML model, right click the editor and choose "Open Diagram"). On the other hand, the Eclipse extension allows using the Sirius framework to create custom OML diagrams (although a default set of OML diagrams are bundled and can be used as is). To enable this, right click on your OML project in Rosetta and select Configure -> Convert to Modeling Project. Then right click one more time and choose Viewpoint Selection (at the top) and select OML Vocabulary and Descriiption viewpoints. Then, expand any of your oml diagrams in the Project Explorer view and right click on the Vocabulary or Description element, and select Create New Diagram. This will create/open a Sirius diagram editor. 

A generic Sirius tutorial can be followed [here](https://wiki.eclipse.org/Sirius/Tutorials/StarterTutorial). I plan to create a similar one for OML.
